### DISCUSSION TITLE
<title>how to remove a commit

### DISCUSSION THREAD

**Discussion by Gloriaokewu0194** [community/community#41912] [Projects and Issues]
### Select Topic Area

General

### Body

i created a commit on my github and it is refusing to delect or revert. i need help on how to delect it

**Comment by mohamedzhioua** [comment]
take a look here 
https://stackoverflow.com/questions/448919/how-can-i-remove-a-commit-on-github?answertab=trending#tab-top

**Comment by mohamedzhioua** [comment]
this one can help too 
https://stackoverflow.com/questions/45403222/how-to-completely-delete-a-commit-from-github

**Comment by jfullstackdev** [comment]
do you use Git ? GitHub Desktop ? or directly there on GitHub ? 

if you edited that directly on GitHub but without a PR, then you don't have the revert button

**Comment by airtower-luna** [comment]
For how to edit commit history see [Git Tools - Rewriting History](https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History). A few things to keep in mind:

* After making the modification you will have to force-push to replace the existing history. This is a potentially destructive operation (you will lose the old history), so double check the modified history is exactly as you want it before force-pushing. :warning: 
* If you have already shared the repository, anyone who already has the old (unmodified) history will have to manually accept the new history (or not). This is generally accepted for development branches, for a default branch people might expect an explanation.
* If you want to [remove sensitive data](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/removing-sensitive-data-from-a-repository) rewriting the history is not enough, you'll have to contact GitHub support after and ask them to clear the caches to make sure the old commit is really gone.
